fastcrypto-tbls/src/nodes.rs

```diff
@@ -76,4 +76,35 @@ impl<G: GroupElement> Nodes<G> {
     pub fn iter(&self) -> impl Iterator<Item = &Node<G>> {
         self.nodes.iter()
     }
+
+    /// Reduce weights up to an allowed delta in the original total weight.
+    /// Finds the largest d such that:
+    /// - The new threshold is ceil(t / d)
+    /// - The new weights are all divided by d (floor division)
+    /// - The precision loss, counted as the sum of the remainders of the division by d, is at most
+    ///   the allowed delta
+    /// In practice, allowed delta will be the extra liveness we would assume above 2f+1.
+    pub fn reduce(&self, t: u16, allowed_delta: u16) -> (Self, u16) {
+        let mut max_d = 1;
+        for d in 2..=50 {
+            let sum = self.nodes.iter().map(|n| n.weight % d).sum::<u16>();
+            if sum <= allowed_delta {
```

Comment:
Maybe I'm misunderstanding this, but even after sum > allowed_delta for some d, there could still be a larger d which makes the sum <= allowed_delta, so I don't think you can quit the loop early here.